Apps/phadldap/adldap_connector.py

```diff
@@ -0,0 +1,783 @@
+            # the next two blocks could be abstracted into a method,
+            # but the author feels that somewhat duplicate code is a small
+            # cost for arguably greater readability.
+
+            # finding groups dn by sam
+            t_group = self._sam_to_dn(groups)   # omit action_result to avoid updating user count
+            for k, v in list(t_group.items()):
+                if v is False:
+                    group_nf.append(k)
+                else:
+                    n_groups.append(v)
+
+            # ensure we actually have a least 1 user and group to modify
+            if len(n_members) > 0 and len(n_groups) > 0:
+                members = n_members
+                groups = n_groups
+            else:
+                self.debug_print("[DEBUG] n_members = {}".format(n_members))
+                self.debug_print("[DEBUG] n_groups = {}".format(n_groups))
+                return RetVal(
+                    action_result.set_status(phantom.APP_ERROR, "Not enough groups or members.")
+                )
+
+        try:
+            if add:
+                func = "added"
+                ldap3.extend.microsoft.addMembersToGroups.ad_add_members_to_groups(
+                    connection=self._ldap_connection,
+                    members_dn=members,
+                    groups_dn=groups,
+                    fix=True,
+                    raise_error=True
+                )
+            else:
+                func = "removed"
+                ldap3.extend.microsoft.removeMembersFromGroups.ad_remove_members_from_groups(
+                    connection=self._ldap_connection,
+                    members_dn=members,
+                    groups_dn=groups,
+                    fix=True,
+                    raise_error=True
+                )
+        except Exception as e:
+            return RetVal(action_result.set_status(
+                phantom.APP_ERROR,
+                "",
+                exception=e
+            ))
+
+        # add action data results
+        for i in members:
+            for j in groups:
+                action_result.add_data({
+                    "member": i,
+                    "group": j,
+                    "function": func
+                })
+
+        return RetVal(action_result.set_status(
+            phantom.APP_SUCCESS,
+            "{} member(s) to group(s).".format(func)
+        ))
+
+    def _handle_unlock_account(self, param):
+        """
+        This method unlocks an active directory account.
+
+        If the use_samaccountname is checked, method will resolve
+        the input samaccountname(user) -> distinguishedName and then use the
+        ldap3 library to unlock the account.
+        """
+        action_result = self.add_action_result(ActionResult(dict(param)))
+        summary = action_result.update_summary({})
+        user = param['user'].lower()
+        ar_data = {}            # data for action_result
+
+        if param.get("use_samaccountname", False):
```

Comment:
small nit: False is not needed as the second parameter, if the key is not found then it already returns False.

Reply:
Nice catch.